### PR TITLE
GH-9611: Add GetChannelMembersByIds to Plugin API

### PR DESCRIPTION
**Pull request by knrt10**
#### Summary
This PR adds a `GetChannelMembersByIds(channelId string, userIds []string)` method to plugin API.

#### Ticket Link
Fixes [#9611](https://github.com/mattermost/mattermost-server/issues/9611)

### CODE REVIEW

File: plugin/api.go [hunk 1]
```diff
@@ -143,6 +143,9 @@ type API interface {
 	// GetChannelMembers gets a channel membership for all users.
 	GetChannelMembers(channelId string, page, perPage int) (*model.ChannelMembers, *model.AppError)
 
+	// GetChannelMembersByIds gets a channel membership for a particualar User
```
Comment (saturninoabril) [inline plugin/api.go]: Spelling check: `particular`

Comment (knrt10) [reply plugin/api.go]: Ok let me update it

File: plugin/api.go [hunk 2]
```diff
@@ -152,6 +152,9 @@ type API interface {
 	// GetChannelMembers gets a channel membership for all users.
 	GetChannelMembers(channelId string, page, perPage int) (*model.ChannelMembers, *model.AppError)
 
+	// GetChannelMembersByIds gets a channel membership for a particular User
```
Comment (hanzei) [inline plugin/api.go]: Can you document the minimum version required here?

```suggestion
	// GetChannelMembersByIds gets a channel membership for a particular User
	//
	// Minimum server version: 5.6
```

**Comment by knrt10** [comment]
@saturninoabril updated.

File: plugin/plugintest/api.go [hunk 3]
```diff
@@ -483,6 +483,7 @@ func (_m *API) GetChannelsForTeamForUser(teamId string, userId string, includeDe
 	return r0, r1
 }
 
+<<<<<<< HEAD
```
Comment (saturninoabril) [inline plugin/plugintest/api.go]: @knrt10 Looks like merge conflicts were not fixed

Comment (knrt10) [reply plugin/plugintest/api.go]: oh sorry, my bad. Updated it now @saturninoabril but I don't understand why build is failing

Comment (knrt10) [reply plugin/plugintest/api.go]: I rebased with master branch

**Comment by cpanato** [comment]
@knrt10 these are the errors in govet

```
plugin/plugintest/example_hello_user_test.go:46:10: cannot use api (type *plugintest.API) as type plugin.API in argument to p.MattermostPlugin.SetAPI:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
# github.com/mattermost/mattermost-server/app [github.com/mattermost/mattermost-server/app.test]
app/plugin_hooks_test.go:336:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
app/plugin_hooks_test.go:429:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
app/plugin_hooks_test.go:481:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
app/plugin_hooks_test.go:530:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
app/plugin_hooks_test.go:577:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
app/plugin_hooks_test.go:645:56: cannot use &mockAPI (type *plugintest.API) as type plugin.API in return argument:
	*plugintest.API does not implement plugin.API (missing GetChannelStats method)
Makefile:254: recipe for target 'govet' failed
```

**Comment by saturninoabril** [comment]
Thanks @knrt10!

**Pull request merged**